lib/heroku_build_report.rb

```diff
@@ -0,0 +1,53 @@
+require 'yaml'
+require 'pathname'
+
+# Observability reporting for builds
+#
+# Example usage:
+#
+#   HerokuBuildReport::GLOBAL.capture(
+#     "ruby_version" => "3.4.2"
+#   )
+module HerokuBuildReport
+  # Accumulates data in memory and writes it to the specified path in YAML format
```

Comment:
When I first read this comment I wondered what happened if the buildpack exited partway through (eg was the report lost since it's "in memory") - but I see that a write occurs as part of every `.capture()` call, so there will be no data loss (and no need for a final writer in a `rescue` somewhere). Is that something we can make clearer in this comment? :-)

Reply:
I added some extra docs